### PR TITLE
Remove `spin plugin update`

### PR DESCRIPTION
**Pull request by kate-goldenring**
fixes https://github.com/fermyon/spin/issues/939

Signed-off-by: Kate Goldenring <kate.goldenring@fermyon.com>

### CODE REVIEW

**Review by itowlson — COMMENTED** [review]
After reading this, and being reminded of how the plugin system works, I'm not sure that trying to achieve consistency between the template and plugin UIs with this particular renaming is desirable.  We risk ending up with two things that _look_ the same but _behave_ very differently.

The templates system has a very simplistic approach in which the installation source (whether Git repo or local directory) is the source of truth: installation means getting whatever is there, with or without overwrite.

Plugins, if memory serves, have a more sophisticated system involving a (optional) centralised catalogue and (I think) a local snapshot of that centralised catalogue.  So there are separate concepts of _updating the local snapshot_ (the update operation, I think?) and _installing a plugin mentioned in the local snapshot_ (the install or upgrade operation, I think?).

So, if I've understood this correctly (and that's a big if), `update` is a catalogue-level operation, whereas `install` is a plugin-level operation.  And mixing the two makes me a bit confused - I asked to `install --update foo` but won't this have an observable side effect if I later install `bar`?

More importantly, if I've understood this right, we risk having two command/options that look the same but behave quite differently.  In templates, `--update` is the _overwrite_ flag - `install` is always fetching directly source, and `--update` controls whether things the user already have are left alone or updated.  In plugins, `--update` is the _sync snapshot_ flag - it doesn't affect whether an existing plugin is retained or overwritten, it just affects _which version_ it gets overwritten with.

To be clear I am not saying either behaviour is wrong or right (or even that we should standardise on one of them).  I'm just saying they're _different_ and that using the same name for them risks misleading users into thinking otherwise.

Sorry for the long explanation, and for recapitulating stuff that I know you know - it wasn't (and isn't) fresh in my mind and I wanted to write down my understanding so it was clear if I was getting it wrong.  Hope my worries sense and that this doesn't come across the wrong way.

**Comment by itowlson** [comment]
Also sorry for not thinking of this during the earlier discussions!

File: src/commands/plugins.rs [hunk 1]
```diff
@@ -205,6 +207,10 @@ pub struct Upgrade {
     /// Allow downgrading a plugin's version.
     #[clap(short = 'd', long = "downgrade", takes_value = false)]
     pub downgrade: bool,
+
+    /// Avoid fetching the latest plugins available to install
```
[the remark this answers is not in the document]
Comment (kate-goldenring) [reply src/commands/plugins.rs:211]: We were modeling after `apt` a little with this. `apt update` updates the package list of what can be updated and `apt upgrade` upgrades all your installed packages to the latest versions: https://www.freecodecamp.org/news/sudo-apt-get-update-vs-upgrade-what-is-the-difference/.

The flag here is whether to automatically do the update part (getting latest) when doing an upgrade. I dont know when you wouldnt want to do this, though, so i made it default to do so

**Comment by kate-goldenring** [comment]
I think we should close this and the related issue https://github.com/fermyon/spin/issues/939#issuecomment-1341602446 and stick with the original intent of plugins matching normal flow of other package management systems

**Pull request closed without merging**